### DISCUSSION TITLE
SnapshotDb.createEmpty for unit testing

### DISCUSSION THREAD

**Discussion by spowell-216** [iTwin/itwinjs-core#1184] [Q&A] [answered]
Hi, 
I am trying to implement some unit testing modeled after some of the tests on your repository.  I am struggling with the Snapshot.createEmpty keeps throwing the following error:

> error thrown: TypeError: Cannot read property 'DgnDb' of undefined

```
import { SnapshotDb } from "@bentley/imodeljs-backend";

( () => {
  try { 
    const fileName: string = 'c:/temp/test.bim';
    const iModelDb = SnapshotDb.createEmpty(fileName, { rootSubject: { name: "testSubject" }, createClassViews: true });
    iModelDb.saveChanges();
    iModelDb.close();
    console.log("done");  
  }
  catch(err){
    console.log(`error thrown: ${err}`);
  }
 })();
```

Is there something else that needs to be initialized prior to creating?

**Comment by pmconne** [comment, marked as answer]
You need to initialize [IModelHost](https://www.itwinjs.org/learning/backend/imodelhost/).
